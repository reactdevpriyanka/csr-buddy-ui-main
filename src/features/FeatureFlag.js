import useAthena from '@/hooks/useAthena';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import useFeature from './useFeature';

const useStyles = makeStyles((theme) => ({
  defaultMsgLayout: {
    width: '100%',
    textAlign: 'center',
    marginTop: '10px',
  },
  defaultMsg: {
    ...theme.fonts.h1,
    color: theme.palette.blue.dark,
    fontWeight: '500',
    paddingLeft: `${theme.utils.fromPx(28)}`,
  },
}));
export default function FeatureFlag({
  flag,
  children,
  fallback = null,
  showDefaultBackup = false,
}) {
  const flagValue = useFeature(flag);
  const { getLang } = useAthena();
  const classes = useStyles();

  if (flagValue) {
    return children;
  }

  if (showDefaultBackup) {
    return (
      <div className={classes.defaultMsgLayout}>
        <span className={classes.defaultMsg}>
          {getLang('featureFlagAccessDenialMsg', {
            fallback: "Sorry, you don't have access this feature",
          })}
        </span>
      </div>
    );
  }
  return fallback;
}

FeatureFlag.propTypes = {
  flag: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  fallback: PropTypes.node,
  showDefaultBackup: PropTypes.bool,
};
