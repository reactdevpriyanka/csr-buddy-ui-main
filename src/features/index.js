// TODO: Export all of the modules of this package
export useFeature from './useFeature';
export FeatureFlag from './FeatureFlag';
