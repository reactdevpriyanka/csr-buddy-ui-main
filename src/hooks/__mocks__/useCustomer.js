import customer from '__mock__/customers/nessa';

const mutate = jest.fn();

const mockUseCustomer = jest.fn(() => {
  return {
    mutate,
    error: null,
    data: customer,
    isValidating: false,
    updateCustomer: (id, data) =>
      new Promise((resolve) => {
        return resolve({ data: {} });
      }),
  };
});

mockUseCustomer.mutate = mutate;

export default mockUseCustomer;
