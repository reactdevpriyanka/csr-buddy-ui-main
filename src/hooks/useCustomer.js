import axios from 'axios';
import useSWR from 'swr';
import { useRouter } from 'next/router';
import { SNACKVARIANTS } from '@components/SnackMessage/SnackMessage';
import { useSnackbar } from 'notistack';

export default function useCustomer(customerId) {
  const router = useRouter();

  const { id } = router.query;

  const queryId = customerId ? customerId : id;

  const { data, error, mutate } = useSWR(
    queryId ? `/api/v1/customer/${queryId}` : null,
    async (url) => axios.get(url).then(({ data }) => data),
    {
      revalidateIfStale: false,
      revalidateOnFocus: false,
      revalidateOnReconnect: false,
      shouldRetryOnError: true,
    },
  );

  const { enqueueSnackbar } = useSnackbar();

  const updateCustomer = async (customerId, customerData, onSuccess) =>
    axios
      .patch(`/api/v1/customer/${customerId}`, customerData)
      .then(() => onSuccess())
      .catch(() => {
        enqueueSnackbar({
          messageHeader: 'Error',
          variant: SNACKVARIANTS.ERROR,
          messageSubheader: 'Failed to Update Customer Details',
        });
      });

  return {
    data,
    error,
    updateCustomer,
    mutate,
  };
}
