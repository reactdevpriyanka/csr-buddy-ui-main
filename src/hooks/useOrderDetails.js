import axios from 'axios';
import useSWR from 'swr';
import useSanitizedRouter from './useSanitizedRouter';

export default function useOrderDetails() {
  const { id } = useSanitizedRouter();
  return useSWR(
    id && `/api/v1/orderdetails?customerId=${id}`,
    async (url) => axios.get(url).then(({ data }) => data),
    { revalidateOnFocus: false },
  );
}
