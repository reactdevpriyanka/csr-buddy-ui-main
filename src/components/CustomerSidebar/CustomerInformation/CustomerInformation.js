import { useCallback, useEffect, useState } from 'react';
import Button from '@material-ui/core/Button';
import Progress from '@material-ui/core/CircularProgress';
import { makeStyles } from '@material-ui/core/styles';
import { capitalize } from '@utils/string';
import useCustomer from '@/hooks/useCustomer';
import usePanels from '@/hooks/usePanels';
import LabelValueField from '@components/LabelValueField';
import Typography from '@material-ui/core/Typography';
import Link from '@mui/material/Link';
import { FeatureFlag } from '@/features';
import { useRouter } from 'next/router';
import ResetPasswordModal from '@components/ResetPasswordModal';
import useScrollLock from '@/hooks/useScrollLock';
import useAthena from '@/hooks/useAthena';
import CustomerAddressBook from '../CustomerOverview/CustomerAddressBook';

const useStyles = makeStyles((theme) => ({
  formContainer: {
    marginTop: theme.spacing(2),
    padding: `0 0 ${theme.spacing(1.1)}`,
    borderBottom: '1px solid #f5f5f5',
  },
  heading: {
    ...theme.fonts.h2,
    marginTop: theme.spacing(0.25),
    marginBottom: theme.spacing(1.2),
    marginRight: theme.spacing(0.5),
    color: theme.palette.blue.dark,
    whiteSpace: 'nowrap',
  },
  headerSpan: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  label: {
    color: '#1C49C2',
    fontWeight: '700',
  },
  link: {
    alignItems: 'flex-end',
    fontWeight: '700',
    fontSize: '14px',
    paddingTop: theme.spacing(0.4),
    cursor: 'pointer',
    textTransform: 'uppercase',
    color: '#1C49C2',
  },
  resetPasswordLink: {
    padding: `0`,
    display: 'block',
    color: theme.palette.blue.chewyBrand,
    border: '1px solid transparent',
    transition: 'all 0.2s',
    textTransform: 'none',
    '&:active': {
      border: `1px solid ${theme.palette.blue.chewyBrand}`,
      padding: `0 ${theme.utils.fromPx(6)}`,
    },
    '&:not(:first-child)': {
      marginTop: theme.utils.fromPx(4),
    },
  },
  giftCardLink: {
    padding: `0`,
    display: 'block',
    color: theme.palette.blue.chewyBrand,
    border: '1px solid transparent',
    transition: 'all 0.2s',
    textTransform: 'none',
    '&:focus': {
      border: `1px solid ${theme.palette.blue.chewyBrand}`,
      padding: `0 ${theme.utils.fromPx(6)}`,
    },
    '&:not(:first-child)': {
      marginTop: theme.utils.fromPx(4),
    },
  },
}));

const CustomerInformation = () => {
  const classes = useStyles();
  const router = useRouter();
  const { getLang } = useAthena();
  const { unlockScroll } = useScrollLock();

  const [openCustomerAddressBook, setOpenCustomerAddressBook] = useState(false);
  const [openResetPasswordDialog, setOpenResetPasswordDialog] = useState(false);

  const handleResetPasswordClick = useCallback(() => {
    setOpenResetPasswordDialog(true);
  }, [setOpenResetPasswordDialog]);

  const navigateToGiftCardsAndPromotions = () => {
    const { id: customerId } = router.query;
    router.push(`/customers/${customerId}/giftcards-promotions`);
  };

  useEffect(() => {
    if (!openCustomerAddressBook) {
      unlockScroll();
    }
  }, [openCustomerAddressBook]);

  useEffect(() => {
    if (!openResetPasswordDialog) {
      unlockScroll();
    }
  }, [openResetPasswordDialog]);

  const { data, error } = useCustomer();
  const { navigateToPanel } = usePanels();

  if (error) {
    return null;
  }

  if (!data) {
    return (
      <div data-testid="loader">
        <Progress />
      </div>
    );
  }

  const customer = {
    id: '',
    addresses: [],
    customerAddresses: [],
    email: '',
    ...data,
  };

  const primaryAddress =
    customer.addresses.length > 0
      ? customer.addresses.find((address) => address.primaryAddress) || customer.addresses[0]
      : null;

  const phoneNumberFormatter = function (phone) {
    const phoneRegex = /(\d{3})(\d{3})(\d{4})/g;

    if (isValidPhoneNumber(phone)) {
      const matches = phoneRegex.exec(phone.replace(/\D/g, ''));
      return '(' + matches[1] + ')' + matches[2] + '-' + matches[3];
    }
    return phone;
  };

  const isValidPhoneNumber = function (value) {
    return value && value.match(/^\(?(\d{3})\)?[ .-]?(\d{3})[ .-]?(\d{4})$/);
  };

  const onEditCustomerInformation = () => {
    navigateToPanel('customerInformationEditor');
  };

  return (
    <>
      <form
        autoComplete="off"
        data-testid="customer-sidebar:form"
        className={classes.formContainer}
      >
        <span className={classes.headerSpan}>
          <Typography className={classes.heading} variant="h2">
            {getLang('customerDetailsText', { fallback: 'Customer Details' })}
          </Typography>
          <FeatureFlag flag="feature.explorer.editCustomerDetailsEnabled">
            <Link
              variant="text"
              color="#1C49C2"
              aria-label="edit customer information"
              component="span"
              data-testid="edit-customer-info"
              underline="hover"
              className={classes.link}
              onClick={onEditCustomerInformation}
            >
              Edit
            </Link>
          </FeatureFlag>
        </span>
        <LabelValueField
          id="cid"
          name="cid"
          label="CID:"
          data-testid="customer:cid"
          value={customer.id}
        />
        <LabelValueField
          id="status"
          name="status"
          label="Status:"
          data-testid="customer:status"
          value={customer.status}
          valueColor={customer.status === 'ACTIVE' ? '#0A8E4E' : '#851940'}
        />
        <LabelValueField
          id="email"
          name="email"
          label="Email:"
          data-testid="customer:email"
          value={customer.email}
        />
        <LabelValueField
          id="phoneNumber"
          name="phoneNumber"
          label="Phone:"
          data-testid="customer:phoneNumber"
          value={primaryAddress?.phone}
          valueFormatter={phoneNumberFormatter}
        />
        <LabelValueField
          id="location"
          name="location"
          label="Location:"
          data-testid="customer:location"
          value={
            primaryAddress?.city && primaryAddress?.state
              ? `${capitalize(primaryAddress?.city)}, ${primaryAddress?.state}`
              : ''
          }
        />
        <FeatureFlag flag="feature.explorer.resetPasswordEnabled">
          <Button
            variant="text"
            onClick={handleResetPasswordClick}
            data-testid="reset-password-link"
            className={classes.resetPasswordLink}
          >
            {'Reset Password'}
          </Button>
          <ResetPasswordModal
            isOpen={openResetPasswordDialog}
            setOpenDialog={setOpenResetPasswordDialog}
          />
        </FeatureFlag>
        <FeatureFlag flag="feature.explorer.customerAddressBookEnabled">
          <Button
            variant="text"
            onClick={() => setOpenCustomerAddressBook(true)}
            data-testid="customer-addressbook-icon"
            className={classes.resetPasswordLink}
          >
            {'Address Book'}
          </Button>
        </FeatureFlag>
        <FeatureFlag flag="feature.explorer.giftCardsAndPromotionsPageEnabled">
          <Button
            variant="text"
            onClick={navigateToGiftCardsAndPromotions}
            data-testid="gift-cards-and-promotions-link"
            className={classes.giftCardLink}
          >
            {'Gift Cards & Promotions'}
          </Button>
        </FeatureFlag>
      </form>
      {openCustomerAddressBook && (
        <CustomerAddressBook
          isOpen={openCustomerAddressBook}
          openDialog={setOpenCustomerAddressBook}
          customer={customer}
        />
      )}
    </>
  );
};

CustomerInformation.propTypes = {};

export default CustomerInformation;
