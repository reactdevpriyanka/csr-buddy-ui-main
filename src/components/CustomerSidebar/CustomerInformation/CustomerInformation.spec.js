import { renderWrap } from '@utils';
import useCustomer from '@/hooks/useCustomer';
import customerNoPrimaryAddress from '__mock__/customers/nessa+noprimaryaddress';
import customerNoAddress from '__mock__/customers/nessa+noaddress';
import customerNoEmail from '__mock__/customers/nessa+noemail';
import useOrderDetails from '@/hooks/useOrderDetails';
import { useRouter } from 'next/router';
import customerData from '__mock__/customers/happy';
import * as features from '@/features';
import * as nextRouter from 'next/router';
import { waitFor } from '@testing-library/dom';
import CustomerInformation from './CustomerInformation';

jest.mock('@/hooks/useCustomer');
jest.mock('@/hooks/useOrderDetails');
jest.mock('@/hooks/useEnactment');
jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

window.bttUT = {
  start: jest.fn(),
  end: jest.fn(),
};

describe('<CustomerInformation />', () => {
  const render = renderWrap(CustomerInformation);
  const mockRouter = jest.spyOn(nextRouter, 'useRouter');

  mockRouter.mockReturnValue({
    query: {
      id: '123123',
    },
  });

  test('it should render the customer information', () => {
    expect(render()).toMatchSnapshot();
  });

  describe('when customer has no primary address', () => {
    describe('and no addresses on file', () => {
      beforeEach(() => {
        useCustomer.mockReturnValue({
          data: customerNoAddress,
          error: null,
        });
      });

      test('it should render no location', () => {
        const { getByTestId } = render();
        expect(getByTestId('customer:location:label')).toHaveTextContent('Location:');
        expect(getByTestId('customer:location:value')).toHaveTextContent('');
      });
    });

    describe('but has at least one address', () => {
      beforeEach(() => {
        useCustomer.mockReturnValue({
          data: customerNoPrimaryAddress,
          error: null,
        });
      });

      test('it should render first address on file', () => {
        const { getByTestId } = render();
        expect(getByTestId('customer:location:value')).toHaveTextContent('Hollywood, FL');
        expect(getByTestId('customer:phoneNumber:value')).toHaveTextContent('(555)456-3320');
      });
    });
  });

  describe('when customer is loading', () => {
    beforeAll(() => {
      useCustomer.mockReturnValue({
        error: null,
        data: null,
      });
    });

    test('it should render a loader', () => {
      const { getByTestId } = render();
      expect(getByTestId('loader')).toBeInTheDocument();
    });
  });

  describe('when error has occurred', () => {
    beforeAll(() => {
      useCustomer.mockReturnValue({
        error: 'Some nasty bug',
        data: null,
      });
    });

    test('it should render nothing', () => {
      const { container } = render();
      expect(container.childNodes).toHaveLength(0);
    });
  });

  describe('with missing email', () => {
    beforeAll(() => {
      useCustomer.mockReturnValue({
        data: customerNoEmail,
        error: null,
      });
    });

    test('it should render empty email field', () => {
      const { getByTestId } = render();
      expect(getByTestId('customer:email:value')).toHaveTextContent('');
    });
  });

  describe('with valid email', () => {
    beforeAll(() => {
      useCustomer.mockReturnValue({
        data: customerNoPrimaryAddress,
        error: null,
      });
    });

    test('it should render email field', () => {
      const { getByTestId } = render();
      expect(getByTestId('customer:email:value')).toHaveTextContent('nbobbins@chewy.net');
    });
  });

  describe('Should show address book button', () => {
    beforeEach(() => {
      useCustomer.mockReturnValue({
        data: customerData,
        error: null,
      });
      useOrderDetails.mockReturnValue({
        error: null,
        data: {
          ordersFor12MonthRollingPeriod: 20,
        },
      });
      useRouter.mockReturnValue({
        query: {
          id: '123123',
        },
      });
    });

    test('it should show address book button', () => {
      jest.spyOn(features, 'useFeature').mockImplementation(() => true);
      const { getByTestId } = render();
      expect(getByTestId('customer-addressbook-icon')).toBeInTheDocument();
    });

    test('it should not show address book icon if feature flag is off', () => {
      jest.spyOn(features, 'useFeature').mockImplementation(() => false);
      const { queryByTestId } = render();
      expect(queryByTestId('customer-addressbook-icon')).not.toBeInTheDocument();
    });

    test('it should show address book when icon is clicked', () => {
      jest.spyOn(features, 'useFeature').mockImplementation(() => true);
      const { getByTestId } = render();
      expect(getByTestId('customer-addressbook-icon')).toBeInTheDocument();
      render.trigger.click(getByTestId('customer-addressbook-icon'));
      expect(getByTestId('customer-addressbook-dialog')).toBeInTheDocument();
    });
  });

  it('edit pressed pushes customerInformationEditor to router', async () => {
    useCustomer.mockReturnValue({
      data: customerData,
      error: null,
    });

    const pushMock = jest.fn();

    useRouter.mockReturnValue({
      query: {
        id: '123123',
      },
      push: pushMock,
    });

    const { getByTestId } = render();

    render.trigger.click(getByTestId('edit-customer-info'));

    await waitFor(() => expect(pushMock).toHaveBeenCalled());

    expect(pushMock.mock.calls[0][0]).toMatchObject({
      query: {
        id: '123123',
        interactionPanel: 'customerInformationEditor',
      },
    });
  });

  it('gift card links route to /customers/:customerId/giftcards-promotions', async () => {
    useCustomer.mockReturnValue({
      data: customerData,
      error: null,
    });

    const pushMock = jest.fn();

    useRouter.mockReturnValue({
      query: {
        id: '123123',
      },
      push: pushMock,
    });

    const { getByTestId } = render();

    render.trigger.click(getByTestId('gift-cards-and-promotions-link'));

    await waitFor(() => expect(pushMock).toHaveBeenCalled());

    expect(pushMock).toHaveBeenCalledWith(`/customers/123123/giftcards-promotions`);
  });

  it('reset password click launches reset password modal', () => {
    useCustomer.mockReturnValue({
      data: customerData,
      error: null,
    });

    useRouter.mockReturnValue({
      query: {
        id: '123123',
      },
    });

    const { queryByTestId, getByTestId } = render();

    render.trigger.click(getByTestId('reset-password-link'));

    expect(queryByTestId('reset-password-modal')).toBeInTheDocument();
  });
});
