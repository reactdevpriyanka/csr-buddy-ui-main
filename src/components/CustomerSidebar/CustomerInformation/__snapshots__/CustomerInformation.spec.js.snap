// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomerInformation /> it should render the customer information 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        autocomplete="off"
        class="makeStyles-formContainer-1"
        data-testid="customer-sidebar:form"
      >
        <span
          class="makeStyles-headerSpan-3"
        >
          <h2
            class="MuiTypography-root makeStyles-heading-2 MuiTypography-h2"
          >
            Customer Details
          </h2>
          <span
            aria-label="edit customer information"
            class="MuiTypography-root MuiTypography-text MuiLink-root MuiLink-underlineHover makeStyles-link-5 css-1ifudzl-MuiTypography-root-MuiLink-root"
            data-testid="edit-customer-info"
          >
            Edit
          </span>
        </span>
        <div
          class="makeStyles-root-10"
        >
          <span>
            <label
              class="MuiFormLabel-root makeStyles-label-11"
              data-testid="customer:cid:label"
            >
              CID:
            </label>
          </span>
          <span>
            <label
              class="MuiFormLabel-root makeStyles-value-12 makeStyles-value-17"
            >
              <span
                class="makeStyles-valueText-13"
                data-testid="customer:cid:value"
              >
                148824357
              </span>
              <span
                class="makeStyles-copiedSpan-14"
                hidden=""
              >
                <span>
                  <span />
                </span>
                <span
                  class="makeStyles-copiedText-16"
                  data-testid="customer:cid:copied"
                  hidden=""
                >
                  COPIED
                </span>
              </span>
            </label>
          </span>
        </div>
        <div
          class="makeStyles-root-10"
        >
          <span>
            <label
              class="MuiFormLabel-root makeStyles-label-11"
              data-testid="customer:status:label"
            >
              Status:
            </label>
          </span>
          <span>
            <label
              class="MuiFormLabel-root makeStyles-value-12 makeStyles-value-18"
            >
              <span
                class="makeStyles-valueText-13"
                data-testid="customer:status:value"
              />
              <span
                class="makeStyles-copiedSpan-14"
                hidden=""
              >
                <span>
                  <span />
                </span>
                <span
                  class="makeStyles-copiedText-16"
                  data-testid="customer:status:copied"
                  hidden=""
                >
                  COPIED
                </span>
              </span>
            </label>
          </span>
        </div>
        <div
          class="makeStyles-root-10"
        >
          <span>
            <label
              class="MuiFormLabel-root makeStyles-label-11"
              data-testid="customer:email:label"
            >
              Email:
            </label>
          </span>
          <span>
            <label
              class="MuiFormLabel-root makeStyles-value-12 makeStyles-value-19"
            >
              <span
                class="makeStyles-valueText-13"
                data-testid="customer:email:value"
              >
                nbobbins@chewy.net
              </span>
              <span
                class="makeStyles-copiedSpan-14"
                hidden=""
              >
                <span>
                  <span />
                </span>
                <span
                  class="makeStyles-copiedText-16"
                  data-testid="customer:email:copied"
                  hidden=""
                >
                  COPIED
                </span>
              </span>
            </label>
          </span>
        </div>
        <div
          class="makeStyles-root-10"
        >
          <span>
            <label
              class="MuiFormLabel-root makeStyles-label-11"
              data-testid="customer:phoneNumber:label"
            >
              Phone:
            </label>
          </span>
          <span>
            <label
              class="MuiFormLabel-root makeStyles-value-12 makeStyles-value-20"
            >
              <span
                class="makeStyles-valueText-13"
                data-testid="customer:phoneNumber:value"
              >
                (555)456-3320
              </span>
              <span
                class="makeStyles-copiedSpan-14"
                hidden=""
              >
                <span>
                  <span />
                </span>
                <span
                  class="makeStyles-copiedText-16"
                  data-testid="customer:phoneNumber:copied"
                  hidden=""
                >
                  COPIED
                </span>
              </span>
            </label>
          </span>
        </div>
        <div
          class="makeStyles-root-10"
        >
          <span>
            <label
              class="MuiFormLabel-root makeStyles-label-11"
              data-testid="customer:location:label"
            >
              Location:
            </label>
          </span>
          <span>
            <label
              class="MuiFormLabel-root makeStyles-value-12 makeStyles-value-21"
            >
              <span
                class="makeStyles-valueText-13"
                data-testid="customer:location:value"
              >
                Hollywood, FL
              </span>
              <span
                class="makeStyles-copiedSpan-14"
                hidden=""
              >
                <span>
                  <span />
                </span>
                <span
                  class="makeStyles-copiedText-16"
                  data-testid="customer:location:copied"
                  hidden=""
                >
                  COPIED
                </span>
              </span>
            </label>
          </span>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-resetPasswordLink-6"
          data-testid="reset-password-link"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Reset Password
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-resetPasswordLink-6"
          data-testid="customer-addressbook-icon"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Address Book
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-giftCardLink-7"
          data-testid="gift-cards-and-promotions-link"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Gift Cards & Promotions
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      autocomplete="off"
      class="makeStyles-formContainer-1"
      data-testid="customer-sidebar:form"
    >
      <span
        class="makeStyles-headerSpan-3"
      >
        <h2
          class="MuiTypography-root makeStyles-heading-2 MuiTypography-h2"
        >
          Customer Details
        </h2>
        <span
          aria-label="edit customer information"
          class="MuiTypography-root MuiTypography-text MuiLink-root MuiLink-underlineHover makeStyles-link-5 css-1ifudzl-MuiTypography-root-MuiLink-root"
          data-testid="edit-customer-info"
        >
          Edit
        </span>
      </span>
      <div
        class="makeStyles-root-10"
      >
        <span>
          <label
            class="MuiFormLabel-root makeStyles-label-11"
            data-testid="customer:cid:label"
          >
            CID:
          </label>
        </span>
        <span>
          <label
            class="MuiFormLabel-root makeStyles-value-12 makeStyles-value-17"
          >
            <span
              class="makeStyles-valueText-13"
              data-testid="customer:cid:value"
            >
              148824357
            </span>
            <span
              class="makeStyles-copiedSpan-14"
              hidden=""
            >
              <span>
                <span />
              </span>
              <span
                class="makeStyles-copiedText-16"
                data-testid="customer:cid:copied"
                hidden=""
              >
                COPIED
              </span>
            </span>
          </label>
        </span>
      </div>
      <div
        class="makeStyles-root-10"
      >
        <span>
          <label
            class="MuiFormLabel-root makeStyles-label-11"
            data-testid="customer:status:label"
          >
            Status:
          </label>
        </span>
        <span>
          <label
            class="MuiFormLabel-root makeStyles-value-12 makeStyles-value-18"
          >
            <span
              class="makeStyles-valueText-13"
              data-testid="customer:status:value"
            />
            <span
              class="makeStyles-copiedSpan-14"
              hidden=""
            >
              <span>
                <span />
              </span>
              <span
                class="makeStyles-copiedText-16"
                data-testid="customer:status:copied"
                hidden=""
              >
                COPIED
              </span>
            </span>
          </label>
        </span>
      </div>
      <div
        class="makeStyles-root-10"
      >
        <span>
          <label
            class="MuiFormLabel-root makeStyles-label-11"
            data-testid="customer:email:label"
          >
            Email:
          </label>
        </span>
        <span>
          <label
            class="MuiFormLabel-root makeStyles-value-12 makeStyles-value-19"
          >
            <span
              class="makeStyles-valueText-13"
              data-testid="customer:email:value"
            >
              nbobbins@chewy.net
            </span>
            <span
              class="makeStyles-copiedSpan-14"
              hidden=""
            >
              <span>
                <span />
              </span>
              <span
                class="makeStyles-copiedText-16"
                data-testid="customer:email:copied"
                hidden=""
              >
                COPIED
              </span>
            </span>
          </label>
        </span>
      </div>
      <div
        class="makeStyles-root-10"
      >
        <span>
          <label
            class="MuiFormLabel-root makeStyles-label-11"
            data-testid="customer:phoneNumber:label"
          >
            Phone:
          </label>
        </span>
        <span>
          <label
            class="MuiFormLabel-root makeStyles-value-12 makeStyles-value-20"
          >
            <span
              class="makeStyles-valueText-13"
              data-testid="customer:phoneNumber:value"
            >
              (555)456-3320
            </span>
            <span
              class="makeStyles-copiedSpan-14"
              hidden=""
            >
              <span>
                <span />
              </span>
              <span
                class="makeStyles-copiedText-16"
                data-testid="customer:phoneNumber:copied"
                hidden=""
              >
                COPIED
              </span>
            </span>
          </label>
        </span>
      </div>
      <div
        class="makeStyles-root-10"
      >
        <span>
          <label
            class="MuiFormLabel-root makeStyles-label-11"
            data-testid="customer:location:label"
          >
            Location:
          </label>
        </span>
        <span>
          <label
            class="MuiFormLabel-root makeStyles-value-12 makeStyles-value-21"
          >
            <span
              class="makeStyles-valueText-13"
              data-testid="customer:location:value"
            >
              Hollywood, FL
            </span>
            <span
              class="makeStyles-copiedSpan-14"
              hidden=""
            >
              <span>
                <span />
              </span>
              <span
                class="makeStyles-copiedText-16"
                data-testid="customer:location:copied"
                hidden=""
              >
                COPIED
              </span>
            </span>
          </label>
        </span>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-resetPasswordLink-6"
        data-testid="reset-password-link"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Reset Password
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-resetPasswordLink-6"
        data-testid="customer-addressbook-icon"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Address Book
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-giftCardLink-7"
        data-testid="gift-cards-and-promotions-link"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Gift Cards & Promotions
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
